src/main/java/com/blackducksoftware/integration/hub/alert/web/controller/EmailConfigController.java

```diff
@@ -40,87 +37,57 @@
-            return ResponseEntity.created(uri).build();
-        }
-        return ResponseEntity.badRequest().body("No configuration with id " + emailConfig.getId());
+        return super.putConfig(emailConfig);
     }
 
     @Override
     @DeleteMapping(value = "/configuration/email")
     public ResponseEntity<String> deleteConfig(@RequestAttribute(value = "emailConfig", required = true) @RequestBody final EmailConfigRestModel emailConfig) {
-        if (emailConfig.getId() != null && emailRepository.exists(emailConfig.getId())) {
-            emailRepository.delete(emailConfig.getId());
-            return ResponseEntity.ok().build();
-        }
-        return ResponseEntity.badRequest().body("No configuration with id " + emailConfig.getId());
+        return super.deleteConfig(emailConfig);
     }
 
     @Override
     @PostMapping(value = "/configuration/email/test")
     public ResponseEntity<String> testConfig(@RequestAttribute(value = "emailConfig", required = true) final EmailConfigRestModel emailConfig) {
-        // TODO implement method for testing the configuration
-        return ResponseEntity.notFound().build();
+        final EmailChannel channel = new EmailChannel(null, null, (EmailRepository) repository);
+        final String responseMessage = channel.testMessage(restModelToDatabaseModel(emailConfig));
+        return super.createResponse(HttpStatus.OK, emailConfig.getId(), responseMessage);
     }
 
     @Override
     public EmailConfigEntity restModelToDatabaseModel(final EmailConfigRestModel restModel) {
-        final EmailConfigEntity databaseModel = new EmailConfigEntity(restModel.getMailSmtpHost(), restModel.getMailSmtpUser(), restModel.getMailSmtpPassword(), restModel.getMailSmtpPort(), restModel.getMailSmtpConnectionTimeout(),
-                restModel.getMailSmtpTimeout(), restModel.getMailSmtpFrom(), restModel.getMailSmtpLocalhost(), restModel.getMailSmtpEhlo(), restModel.getMailSmtpAuth(), restModel.getMailSmtpDnsNotify(), restModel.getMailSmtpDsnRet(),
-                restModel.getMailSmtpAllow8bitmime(), restModel.getMailSmtpSendPartial(), restModel.getEmailTemplateDirectory(), restModel.getEmailTemplateLogoImage(), restModel.getEmailSubjectLine());
+        final EmailConfigEntity databaseModel = new EmailConfigEntity(restModel.getId(), restModel.getMailSmtpHost(), restModel.getMailSmtpUser(), restModel.getMailSmtpPassword(), restModel.getMailSmtpPort(),
+                restModel.getMailSmtpConnectionTimeout(), restModel.getMailSmtpTimeout(), restModel.getMailSmtpFrom(), restModel.getMailSmtpLocalhost(), restModel.getMailSmtpEhlo(), restModel.getMailSmtpAuth(),
+                restModel.getMailSmtpDnsNotify(), restModel.getMailSmtpDsnRet(), restModel.getMailSmtpAllow8bitmime(), restModel.getMailSmtpSendPartial(), restModel.getEmailTemplateDirectory(), restModel.getEmailTemplateLogoImage(),
+                restModel.getEmailSubjectLine());
         return databaseModel;
     }
 
```

Comment:
The databaseModelsToRestModels method below this can be moved into the parent class as it just loops and calls the above method.

Reply:
Resolved